addons/mail/static/src/views/fields/statusbar_time/statusbar_time_field.js

```diff
@@ -0,0 +1,114 @@
+/** @odoo-module **/
+
+import { _lt } from "@web/core/l10n/translation";
+import {
+    preloadStatusBar,
+    statusBarField,
+    StatusBarField,
+} from "@web/views/fields/statusbar/statusbar_field";
+import { registry } from "@web/core/registry";
+
+export class StatusBarTimeField extends StatusBarField {
+    static template = "web.StatusBarTimeField";
+
+    humanizeDuration(seconds, showFullDuration) {
+        if (seconds < 60) {
+            seconds = 60;
+        }
+        const minuteInSeconds = 60;
+        const hourInSeconds = 60 * minuteInSeconds;
+        const dayInSeconds = 24 * hourInSeconds;
+        const monthInSeconds = 30 * dayInSeconds;
+        const yearInSeconds = 365 * dayInSeconds;
+        const numberOfValues = showFullDuration ? 2 : 1;
+        const displayStyle = showFullDuration ? "long" : "narrow";
+
+        const years = Math.floor(seconds / yearInSeconds);
+        seconds -= years * yearInSeconds;
+        const months = Math.floor(seconds / monthInSeconds);
+        seconds -= months * monthInSeconds;
+        const days = Math.floor(seconds / dayInSeconds);
+        seconds -= days * dayInSeconds;
+        const hours = Math.floor(seconds / hourInSeconds);
+        seconds -= hours * hourInSeconds;
+        const minutes = Math.floor(seconds / minuteInSeconds);
+        seconds -= minutes * minuteInSeconds;
+        const durationObject = {};
+        if (years) {
+            durationObject.years = years;
+        }
+        if (months && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.months = months;
+        }
+        if (days && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.days = days;
+        }
+        if (hours && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.hours = hours;
+        }
+        if (minutes && Object.keys(durationObject).length < numberOfValues) {
+            durationObject.minutes = minutes;
+        }
+        if (Object.keys(durationObject).length === 0) {
+            return false;
+        }
+        const duration = luxon.Duration.fromObject(durationObject);
+        let durationStr = duration.toHuman({ unitDisplay: displayStyle });
+
+        // In English, the symbol for minutes and months is the same (m), so change months to M
+        if (!showFullDuration && duration.loc.locale.includes("en") && durationObject.months > 0) {
```

Comment:
What ? Isn't it a clean way to do it :D ?

Reply:
I wish and spent some time on it but couldn't find a way

Jeh looked into it and using this luxon object this was the cleanest way to display a human-readable time, with luxon handling translations and working well in a variety of languages, so users don't have to manually translate

However, these abbreviations look like they're hardcoded with the locale, so I couldn't find another way than just "replace" in the string

Reply:
Well ... that's a thing :D . And it means by default it messes seconds and months ?